freshmaker/pyxis.py

```diff
@@ -178,21 +178,30 @@ def get_latest_bundles(self, index_images):
 
         return latest_bundles
 
-    def get_manifest_list_digest_by_nvr(self, nvr):
+    def get_manifest_list_digest_by_nvr(self, nvr, must_be_published=True):
         """
         Get image's digest(manifest_list_digest field) by its NVR
 
         :param str nvr: NVR of ContainerImage to query Pyxis
+        :param bool must_be_published: determines if the image must be published to the repository
+            that the manifest list digest is retrieved from; if ``False``, the published
+            manifest list digest is still preferred when present
         :return: digest of image or None if manifest_list_digest not exists
         :rtype: str or None
         """
         request_params = {'include': ','.join(['data.brew', 'data.repositories'])}
 
         # get manifest_list_digest of ContainerImage from Pyxis
         for image in self._pagination(f'images/nvr/{nvr}', request_params):
-            for repo in image.get('repositories'):
+            for repo in image['repositories']:
                 if repo['published'] and 'manifest_list_digest' in repo:
                     return repo['manifest_list_digest']
+            # If no published manifest list digest exists, check if there is an unpublished one
+            # if must_be_published is False
+            if not must_be_published:
+                for repo in image['repositories']:
+                    if 'manifest_list_digest' in repo:
+                        return repo['manifest_list_digest']
```

Comment:
If the `manifest_list_digest` are same, I think we are safe to do this:
```suggestion
                if must_be_published and not repo['published']:
                    continue
                if 'manifest_list_digest' in repo:
                    return repo['manifest_list_digest']
```

Reply:
I was being cautious so I went the other route, but you're right, there's no reason to be so cautious.